qa/smoke-test-client/src/test/java/org/elasticsearch/smoketest/ESSmokeClientTestCase.java

```diff
@@ -81,9 +82,19 @@ private static Client startClient(Path tempDir, TransportAddress... transportAdd
             .put("client.transport.ignore_cluster_name", true)
             .put(Environment.PATH_HOME_SETTING.getKey(), tempDir);
         final Collection<Class<? extends Plugin>> plugins;
+        boolean usNio = random().nextBoolean();
```

Comment:
I think that we should test it for the life of 6.x because it will be extra exercise for NIO.